subxt/src/backend/mod.rs

```diff
@@ -734,4 +872,599 @@ mod test {
+                                let item = data.pop("chainHead_v1_storage".into());
+                                if let Some(sub) = sub {
+                                    let item = item;
+                                    sub.write_delayed(item).await
+                                }
+                            }
+                            response.map(|x| RawValue::from_string(x).unwrap())
+                        })
+                    }),
+                )
+                .add_mock_data(mock_data)
+                .add_mock_data(response_data)
+                .build();
+            let backend = run_backend_spawn_background(rpc_client);
+
+            // We try again and should succeed
+            let response = backend
+                .storage_fetch_values(
+                    ["ID1".into(), "ID2".into(), "ID3".into()].into(),
+                    crate::utils::H256::random(),
+                )
+                .await
+                .unwrap();
+
+            let response = response
+                .map(|x| x.unwrap())
+                .collect::<Vec<StorageResponse>>()
+                .await;
+
+            assert_eq!(
+                vec![
+                    storage_response("ID1", "Data1"),
+                    storage_response("ID2", "Data2"),
+                    storage_response("ID3", "Data3"),
+                ],
+                response
+            )
+        }
+        #[tokio::test]
+        /// Tests the error behaviour in streams based on chainHead_v1_continue calls
+        async fn storage_fetch_values_retry_continue() {
+            fn compare_storage_responses(
+                expected_response: &StorageResponse,
+                received_response: &StorageResponse,
+            ) -> bool {
+                expected_response == received_response
+            }
+
+            let response_data = vec![
+                (
+                    "method_response",
+                    Message::Single(Err::<MethodResponse, RpcError>(
+                        RpcError::DisconnectedWillReconnect("Error".into()),
+                    )),
+                ),
+                (
+                    "method_response",
+                    Message::Single(Ok(response_started("Id1"))),
+                ),
+                (
+                    "method_response",
+                    Message::Single(Ok(response_started("Id1"))),
+                ),
+            ];
+            let continue_data = vec![
+                ("continue_response", Message::Single(Ok(()))),
+                ("continue_response", Message::Single(Ok(()))),
+                (
+                    "continue_response",
+                    Message::Single(Err(RpcError::DisconnectedWillReconnect("Error".into()))),
+                ),
+                ("continue_response", Message::Single(Ok(()))),
+                ("continue_response", Message::Single(Ok(()))),
+            ];
+            let mock_data = vec![
```

Comment:
yeah, ok I see why you separate these now but to follow how they relate to chainHead api is a bit tricky to follow.

It's not really easy for me to follow how the responses is linked the "follow_event/subscription data`,
So I wonder whether we provide something like:

```rust
(
     method_response(id1), 
     Message::Many(vec![Ok(storage_items("Id1", &[storage_result("ID1", "Data1")])), Ok(operation_continue("Id1"))])
)

```

where the "first item" is the method_response i.e, operationStarted and the "second item" is the subscription data i.e, follow_events on provided on that method response.

That way it simpler to understand the responses for each call but it may not worked that well with your test structure

Reply:
hmm, given we know the exact ordering of the requests/responses/subscription messages i can try to refactor it to be one single stack of responses that will be popped one by one. But maybe dong this in a followup PR would be more appropriate?

Reply:
Yeah it's fine to keep it out of this PR, let's hear what James and Alex thinks about this as well :)

Reply:
Yep it would be a bit easier to follow the test with Niklas suggestion. If it turns out too complicated to change it we can do a followup for sure :D

Reply:
Ok, so after a bit of experimenting it looks like it's not possible without adding an external dependency or adding mock data item by item to the mock data set. using `dyn Serialize` is not possible(not object safe) and adding another dependency just to make tests seem a bit more pretty feels like an overkill.